build.template

```diff
@@ -1,7 +1,112 @@
-//! Configuration settings for this Rhai build
+//! Fixed hashing seeds for stable hashing.
 //!
-//! This file is auto-generated from `build.template`
+//! Set to [`None`] to disable stable hashing.
+//!
+//! See [`set_rhai_ahash_seed`] for more.
+//!
+//! Alternatively, You can also set this at compile time by disabling `stable_hash` and setting the `RHAI_AHASH_SEED`
+//! environment variable instead.
+//!
+//! E.g. `env RHAI_AHASH_SEED ="[236,800,954,213]"`
+// [236,800,954,213], haha funny yume nikki reference epic uboachan face numberworld nexus moment 100
+
+use core::{
+    cell::UnsafeCell,
+    marker::PhantomData,
+    mem::MaybeUninit,
+    panic::{RefUnwindSafe, UnwindSafe},
+    sync::atomic::{AtomicBool, Ordering},
+};
+
+// Safety: lol
+struct SusLock<T>
+where
+    T: 'static,
+{
+    initalized: AtomicBool,
+    data: UnsafeCell<MaybeUninit<T>>,
+    _marker: PhantomData<T>,
+}
+
+impl<T> SusLock<T> {
+    pub const fn new() -> SusLock<T> {
+        SusLock {
+            initalized: AtomicBool::new(false),
+            data: UnsafeCell::new(MaybeUninit::uninit()),
+            _marker: PhantomData,
+        }
+    }
+
+    pub fn get(&self) -> Option<&T> {
+        if self.initalized.load(Ordering::SeqCst) {
+            Some(unsafe { (&*self.data.get()).assume_init_ref() })
+        } else {
+            return None;
+        }
+    }
+
+    pub fn get_or_init(&self, f: impl FnOnce() -> T) -> Option<&T> {
+        let value = f();
+        if !self.initalized.load(Ordering::SeqCst) {
+            unsafe {
+                self.data.get().write(MaybeUninit::new(value));
+            }
+            self.initalized.store(true, Ordering::SeqCst);
```

Comment:
I think at least move this _before_ setting `data` to minimize the time between passing the gate and then setting the gate to `true` to block future entries...